src/rgw/driver/rados/rgw_reshard.cc

```diff
@@ -512,6 +572,25 @@ static int init_target_layout(rgw::sal::RadosStore* store,
     store->svc()->bi->clean_index(dpp, bucket_info, target);
     return ret;
   }
+
+  // trim the reshard log entries to guarantee that any existing log entries are cleared,
+  // if there are no reshard log entries, this is a no-op that costs little time
+  if (support_logrecord) {
+    ret = store->getRados()->trim_reshard_log_entries(dpp, bucket_info, null_yield);
+    if (ret == -EOPNOTSUPP) {
+      // not an error, logrecord is not supported, change to block reshard
+      ldpp_dout(dpp, 0) << "WARNING: " << "trim_reshard_log_entries() does not supported"
+                        << " logrecord, falling back to block reshard mode." << dendl;
+      bucket_info.layout.resharding = rgw::BucketReshardState::InProgress;
+      support_logrecord = false;
```

Comment:
`init2` operates dest shards, but `trim_reshard_log_entries()` trims the src shards, whose osds may not contain `cls_method`.

Reply:
right, thanks 👍 

`init_target_layout()` may not be the best place for this part. what do you think about moving this down to `init_reshard()` where we call `set_resharding_status()` on the src shards?

with some extra effort, we might even issue these trim calls in the same osd ops sent by `CLSRGWIssueSetBucketResharding2` there. up to you

Reply:
> moving this down to init_reshard() where we call set_resharding_status() on the src shards

i am working on this

> issue these trim calls in the same osd ops sent by CLSRGWIssueSetBucketResharding2

because `trim_reshard_log_entries()` is also called in `revert_target_layout()`, so I think not making them in the same osd ops in this release.